logger/logger.go

```diff
@@ -107,6 +109,12 @@ func Debug(msg string) {
 	logger.Debug(msg)
 }
 
+func Trace(msg string, attrs ...any) {
+	lock.Lock()
+	defer lock.Unlock()
```

Comment:
Why do we need to grab locks here? I was expecting any (or none) locking to be an implementation specific property.